### PR TITLE
Deprecate setup.py, setup.cfg and use pyproject.toml

### PR DESCRIPTION
**Pull request by cgahr**
1. I excluded test_data/samples/* from being run by black, isort and mypy
2. This allows me to run black, isort and mypy for all files and I can remove the configuration of these 3 tools from tox.ini and github ci. Most importantly, there is no explicit usage of setup.py anymore
3. Since setup.py is not used anymore, I removed it. Its code (as far as I can see) was not necessary.
4. I moved all configurations from setup.cfg to pyproject.toml. 

For a more in-depth discussion see #79

### CODE REVIEW

**Comment by cgahr** [comment]
tox doesn't fail now!

File: tox.ini [hunk 1]
```diff
@@ -12,25 +13,25 @@ deps =
...
     isort
 skip_install = True
 commands =
-    isort --check-only src/ssort tests setup.py
+    isort --check-only .
 
 [testenv:ssort]
 commands =
-    ssort --check --diff src/ssort tests setup.py
+    ssort --check --diff src/ssort tests
```
Comment (bwhmather) [inline tox.ini:27]: We need to fix this.  Tracking in https://github.com/bwhmather/ssort/issues/72.  Surprised pylint and pyflakes haven't implemented something similar.

File: README.rst [hunk 2]
```diff
@@ -104,7 +104,7 @@ We recommend that you reformat using `isort <https://pycqa.github.io/isort/>`_ a
 
 .. code:: bash
 
-    $ ssort src/ tests/ setup.py; isort src/ tests/ setup.py; black src/ tests/ setup.py
```
Comment (bwhmather) [inline README.rst:107]: Remove either all `setup.py`s or none.

**Review by bwhmather — DISMISSED** [review]
I'm happy with this.  Will need to rebase once https://github.com/bwhmather/ssort/pull/80 is in to get CI to pass.

**Comment by jgberry** [comment]
@cgahr I don't think this plays nicely with `.github/workflows/release.yaml`. Probably need some tweaking there.

File: README.rst (hunk 2 above)
Comment (cgahr) [reply README.rst:107]: Missed those; will fix tomorrow

**Comment by cgahr** [comment]
> @cgahr I don't think this plays nicely with `.github/workflows/release.yaml`. Probably need some tweaking there.

You are correct. Will fix this tomorrow.

**Comment by bwhmather** [comment]
> @cgahr I don't think this plays nicely with .github/workflows/release.yaml. Probably need some tweaking there.

Good spot.

**Review by bwhmather — APPROVED** [review]

**Review by jgberry — APPROVED** [review]

**Pull request merged**